pandas/tests/test_window.py

```diff
@@ -1228,242 +1169,75 @@ def roll_mean(x, window, min_periods=None, freq=None, center=False,
-        assert (result[1:] >= 0).all()
-
-        # #2527, ugh
-        arr = np.array([0.00012456, 0.0003, 0])
-        with catch_warnings(record=True):
-            result = mom.rolling_mean(arr, 1)
-        assert result[-1] >= 0
-
-        with catch_warnings(record=True):
-            result = mom.rolling_mean(-arr, 1)
-        assert result[-1] <= 0
-
-    def _check_moment_func(self, f, static_comp, name=None, window=50,
-                           has_min_periods=True, has_center=True,
-                           has_time_rule=True, preserve_nan=True,
-                           fill_value=None, test_stable=False,
-                           zero_min_periods_equal=True,
-                           **kwargs):
-
-        with warnings.catch_warnings(record=True):
-            self._check_ndarray(f, static_comp, window=window,
-                                has_min_periods=has_min_periods,
-                                preserve_nan=preserve_nan,
-                                has_center=has_center, fill_value=fill_value,
-                                test_stable=test_stable,
-                                zero_min_periods_equal=zero_min_periods_equal,
-                                **kwargs)
-
-        with warnings.catch_warnings(record=True):
-            self._check_structures(f, static_comp,
-                                   has_min_periods=has_min_periods,
-                                   has_time_rule=has_time_rule,
-                                   fill_value=fill_value,
-                                   has_center=has_center, **kwargs)
-
-        # new API
-        if name is not None:
-            self._check_structures(f, static_comp, name=name,
-                                   has_min_periods=has_min_periods,
-                                   has_time_rule=has_time_rule,
-                                   fill_value=fill_value,
-                                   has_center=has_center, **kwargs)
-
-    def _check_ndarray(self, f, static_comp, window=50, has_min_periods=True,
-                       preserve_nan=True, has_center=True, fill_value=None,
-                       test_stable=False, test_window=True,
-                       zero_min_periods_equal=True, **kwargs):
-        def get_result(arr, window, min_periods=None, center=False):
-            return f(arr, window, min_periods=min_periods, center=center, **
-                     kwargs)
-
-        result = get_result(self.arr, window)
-        tm.assert_almost_equal(result[-1], static_comp(self.arr[-50:]))
-
-        if preserve_nan:
-            assert (np.isnan(result[self._nan_locs]).all())
-
-        # excluding NaNs correctly
-        arr = randn(50)
-        arr[:10] = np.NaN
-        arr[-10:] = np.NaN
-
-        if has_min_periods:
-            result = get_result(arr, 50, min_periods=30)
-            tm.assert_almost_equal(result[-1], static_comp(arr[10:-10]))
-
-            # min_periods is working correctly
```

Comment:
Do we need to move parts of `_check_ndarray` like this part on checking `min_periods` to `_check_moments_func` ? 

From a superficial look, it seems that `_check_ndarray` did more than `_check_structures` (eg in `_check_structures` I don't see any assertion of NaNs when `min_periods` is specified like here)

I also don't know if `test_stable` and `test_window` make sense for the new interface